src/main/java/cloud/eppo/RuleEvaluator.java

```diff
@@ -0,0 +1,210 @@
+package cloud.eppo;
+
+import static cloud.eppo.Utils.base64Decode;
+import static cloud.eppo.Utils.getMD5Hex;
+
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.OperatorType;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.TargetingCondition;
+import cloud.eppo.ufc.dto.TargetingRule;
+import com.github.zafarkhaja.semver.Version;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class RuleEvaluator {
```

Comment:
Mostly not new code; Largely adapted from Android implementation ([source](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/main/java/cloud/eppo/android/RuleEvaluator.java))